### DISCUSSION TITLE
Nagios force check

### DISCUSSION THREAD

**Discussion by jdmsilbvs** [chriscareycode/nagiostv-react#63] [Ideas]
Hi, is it possible to make a button to force immediate check ?

**Comment by chriscareycode** [comment]
Nice idea. Yes it looks like it can be possible by firing off a specially crafted message to cgi-bin cmd.cgi. Will look into this more when I can get some time.

**Reply by CharlieHird** [reply]
Chris,
I did a little digging and have created this script that passes the information needed for SCHEDULE_FORCED_SVC_CHECK.
(replace // with # in script ...)

//!/bin/sh
// This is a sample shell script showing how you can submit the SCHEDULE_FORCED_SVC_CHECK command
// to Nagios. Adjust variables to fit your environment as necessary.
//
// Edited by Charlie Hird at Shield Advanced Solutions Ltd
// Script forces a reschedule of a service

while getopts "h:s:" optionName; do
case "$optionName" in
h) host=( "$OPTARG" );;
s) service=( "$OPTARG" );;
esac
done


now=`date +%s`
commandfile='/usr/local/nagios/var/rw/nagios.cmd'

/bin/printf "[%lu] SCHEDULE_FORCED_SVC_CHECK;$host;$service;$now\n" $now > $commandfile


Calling this script with -h hostName and -s serviceDescription will force the service to be rescheduled.
I have it tested and working on my install.

Hope this helps.
Cheers,
Charlie
